extendr-api/src/robj/tests.rs

```diff
@@ -156,6 +156,18 @@ fn test_try_from_robj() {
         assert_eq!(<Vec::<Bool>>::try_from(Robj::from(TRUE)), Ok(vec![TRUE]));
         assert_eq!(<Vec::<u8>>::try_from(Robj::from(0_u8)), Ok(vec![0_u8]));
 
+        let v: Result<Real> = r!(NA_REAL).try_into();
+        let mut v: Vec<_> = v.unwrap().collect();
+        assert!(v.pop().unwrap().is_nan());
+        assert_eq!(<Real>::try_from(r!([1.0, 2.0])).unwrap().collect::<Vec<f64>>(), vec![1.0, 2.0]);
+        assert!(<Real>::try_from(r!([true])).is_err());
+
+        assert_eq!(<Int>::try_from(r!([1, 2])).unwrap().collect::<Vec<i32>>(), vec![1, 2]);
```

Comment:
What happens with:
```rust
<Int>::try_from(r!([1.0, 2.0])).unwrap().collect::<Vec<i32>>()
```
Does it work? Often, R numerics will be floats, even if they are meant to store integers.

Reply:
It fails with the appropriate error. This does raise the point that this behavior is not consistent with #243, but that probably should be addressed in the `as_{type}_iter` methods for `Robj`, which this PR's TryFrom conversions use.